src/components/fingerprint/index.js

```diff
@@ -0,0 +1,100 @@
+module.exports = {
+    template: require('fingerprint.html'),
+    data: function() {
+        return {
+	    width: 512,
+	    height: 512,
+	    stream: null,
+        };
+    },
+    props: ['fingerprint', 'friendname', 'context'],
+    created: function() {
+        console.debug('Fingerprint module created!');
+    },
+
+    methods: {
+        close: function() {
+	    if (this.stream != null)
+		this.stream.getVideoTracks()[0].stop();
+            this.$emit("hide-fingerprint");
+        },
+
+	scanQRCode: function() {
+	    console.log("Scan QR code");
+	    if (navigator.mediaDevices && navigator.mediaDevices.getUserMedia) {
+		var that = this;
+		navigator.mediaDevices.getUserMedia({ video: true }).then(function(stream) {
+		    that.stream = stream;
+		    setTimeout(() => {
+			var video = document.getElementById('video');
+			video.srcObject = stream;
+			video.play()
+			that.takeSnapshot(60)
+		    }, 100);
+		}).catch(function(error) {
+		    alert("Is webcam connected?");
```

Comment:
"Unable to start webcam. Check that your webcam is connected and make sure you've granted your browser permission to access it."